lib/settings/settings.js

```diff
@@ -0,0 +1,192 @@
+
+    const addedFiles = context.payload.head_commit.added
+    const modifiedFiles = context.payload.head_commit.modified
+
+    const settingsPath = process.env.SETTINGS_PATH ? process.env.SETTINGS_PATH : 'mergeable.settings.yml'
+    if (!(addedFiles.includes(`.github/${settingsPath}`) || modifiedFiles.includes(`.github/${settingsPath}`))) return
+    const repo = context.repo()
+    if (repo.repo === '.github') {
+      cache.keys().filter(key => key.startsWith(`${repo.owner}/`)).map(key => cache.del(key))
+    } else {
+      cache.del(`${repo.owner}/${repo.repo}/settings`)
+    }
+  }
+
+  static async fetchSettingsFile (context) {
+    let octokit = context.octokit
+    const repo = context.repo()
+
+    // probotContext.settings loads settings from current repo or from a repo called
+    // '.github' in the same organisation as a fallback. It returns the parsed YAML object.
+    let settings = null
+    const settingsPath = process.env.SETTINGS_PATH ? process.env.SETTINGS_PATH : 'mergeable.settings.yml'
+    const cache = Settings.getCache()
+
+    if (process.env.USE_SETTINGS_CACHE !== undefined && process.env.USE_SETTINGS_CACHE === 'true') {
+      Settings.resetCache(context, cache)
+      settings = cache.get(`${repo.owner}/${repo.repo}/settings`)
+      if (settings) {
+        return settings
+      }
+    }
+
+    // First fetch the settings from organization
+    settings = await octokit.repos.getContent({
+      owner: repo.owner,
+      repo: '.github',
+      path: '.github/' + settingsPath
+    }).then(response => {
+      return yaml.safeLoad(Buffer.from(response.data.content, 'base64').toString())
+    }).catch(err => {
+      if (err.status === 404) {
+        return null
+      }
+      throw err
+    })
+
+    // If Organization setting file is null, we will look for the repo
+    if (settings == null) {
+      settings = await octokit.repos.getContent({
+        owner: repo.owner,
+        repo: repo.repo,
+        path: '.github/' + settingsPath
+      }).then(response => {
+        return yaml.safeLoad(Buffer.from(response.data.content, 'base64').toString())
+      }).catch(err => {
+        if (err.status === 404) {
+          return null
+        }
+        throw err
+      })
+    }
+
+    if (_.isUndefined(settings) || _.isNull(settings)) {
+      throw new SettingsFileNotFoundException('Could not find config file.')
+    }
+
+    if (process.env.USE_SETTINGS_CACHE !== undefined && process.env.USE_SETTINGS_CACHE === 'true') {
+      cache.set(`${repo.owner}/${repo.repo}/settings`, settings)
+    }
+
+    return settings
+  }
+
+  static getCache () {
+    return settingsCache
+  }
+
+  static instanceWithContext (context) {
+    return Settings.fetchSettingsFile(context).then(settings => {
+      return new Settings(settings)
+    }).catch(error => {
+      const settings = new Settings()
+      if (error instanceof SettingsFileNotFoundException) {
```

Comment:
if the file is not found, shouldn't u just return default instead of throwing error? Since we are not expecting settings file to always be present

Reply:
I don't understand how this is being resolved, can you explain? i still see that the error is still set. 
Also it is not clear to me the behavior of the app when no settings file is found, can you help me understand?

Reply:
@iammuho can i get an answer for this? I have a feeling this will break production if deployed before it is going to expecting every repo to have settings file

Reply:
Hey @shine2lay ,

I've changed the code a bit for the cases if `mergeable.settings.yml` couldn't be found. Now default options will be passed to cache and context.